### ISSUE TITLE
Expectations regarding DSP work

### ISSUE THREAD

**Issue by andrebrait** [Gericom/GBARunner2#146]
Hi there. I've tested the DSP-enabled build that's now shipping with Twilight Menu++, and it did very nice things for Zelda ALTTP. On my DS, the audio would desynchronize every now and then, whereas it worked perfectly with the DSP-enabled build on my New 3DS. The audio kept in sync and everything was playing perfectly over many minutes of gameplay and many dungeons and many switched between the light and dark worlds.

Do you think this DSP work will be able to (eventually) bring audio compatibility to a near-perfect state for most games?

**Comment by Gericom** [comment]
Yeah, I believe that should be possible. In fact it already works very well in many games. With the dsp I mix the audio by software so I have full control. The only part that is still a bit problematic at times is the fetching of sample data from memory using dma. The priority of this is unfortunately lower than the normal DS dma channels. This means that if a gba game uses large dma transfers with channel 3 (which would normally have a lower priority than the channels used for sound) the dma may be late causing issues.

**Comment by andrebrait** [comment]
Are there any significant differences between the 3DS and the DSi when it comes to DSP?

**Comment by Gericom** [comment]
No, there is no difference regarding the dsp or the software running on it. Aside from that the 3ds has that annoying twlbg glitch that happens more often because of the framerate adjustment.
